server/auth.go

```diff
@@ -367,7 +366,7 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 		noAuthUser string
 	)
 	tlsMap := opts.TLSMap
-	if c.mqtt != nil {
+	if c.isMqtt() {
```

Comment:
Maybe make more generic?

```go
switch c.clientType() {
  case NATS:
  case MQTT:
  case WS:
}
```

Or something like that?

Reply:
Please check commit https://github.com/nats-io/nats-server/pull/1754/commits/2ca0794b48ce10cfa5a14c54ca915595edc1507e when you get a chance. I am not sold on that and I am afraid that it gets mis-used.. meaning you still want to check for c.kind == CLIENT when you deal with routing of messages for instance. So you would need extendedKind() only in specific situations (like the places I use it in this commit).
I can easily revert this commit or have another approach.

Reply:
The enum for what gets returned should include a NON_CLIENT option as well.

Reply:
I guess you miss the fact that this is extendedKind(), not Type() which means that it can be invoked on any type of connection and will return say ROUTER, LEAF, but for a client, it may return CLIENT (if regular) or the new CLIENT_MQTT or CLIENT_WS.
What you suggest (and I do have a version of that) is to have a clientType() that should be invoked only for c.kind == CLIENT and that will then return NATS/MQTT/WS or NON_CLIENT if incorrectly invoked on a non CLIENT connection.

Reply:
Yes the second option is what I had in mind, but minor so..

Reply:
Ok, replaced that commit with 035cffae375e34eed0020aa6f37a533823721601, which I believe does what you asked.